services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java

```diff
@@ -87,71 +87,66 @@ public Response getMainPage() throws IOException, FreeMarkerException {
         URI accountBaseUrl = uriInfo.getBaseUriBuilder().path(RealmsResource.class).path(realm.getName())
                 .path(Constants.ACCOUNT_MANAGEMENT_CLIENT_ID).path("/").build(realm);
 
-        if (!session.getContext().getUri().getRequestUri().getPath().endsWith("/")) {
```

Comment:
@pedroigor @ssilvert I've had to revert the work done under #7794 and #7818 for [KEYCLOAK-17174](https://issues.redhat.com/browse/KEYCLOAK-17174), as the browser router does not add a trailing slash, causing redirects back to the base URL if this code remains.

I don't quite understand the original issue, but it looks like this would no longer affect us. Could I ask you to take a look at the changes in this PR and let me know if it all makes sense?

Reply:
I think the logic to redirect when the path ends with `/` pre-dates that JIRA.

Maybe @pedroigor or @stianst knows why that is done.

If not, I guess we should go ahead as long as the tests are passing.

Reply:
Yeah, looks like you are right. The only thing that I can imagine is that it exists to allow concatenating paths without adding an extra slash (e.g. `accountBaseUrl + "foo/bar"`). But it would be hard to tell where this is being done, or at all.

Reply:
IIRC, the issues you linked are about making sure the URL is built based on the hostname settings. I'll check if we have this covered.

I'm not sure why you had to make all those changes to achieve your goal. Will check it out.

Reply:
I think the white-space formatting is giving the impression that more code has changed. You can change the view of the diff, so that it no longer[ shows these changes](https://github.blog/2018-05-01-ignore-white-space-in-code-review/). The only thing that was removed is the `if` statement and redirect.